src/getIdentifier.ts

```diff
@@ -1,80 +1,81 @@
 }
 
-const DataDnsIdentifier = Record({
-  identifier: String,
-  value: String,
-});
-type DataDnsIdentifier = Static<typeof DataDnsIdentifier>;
-
-const getDnsIdentifierProof = (fragment: VerificationFragment<DataDnsIdentifier | Identity[]>) => {
+const getDnsIdentifierProof = ({ data }: VerificationFragmentWithData<unknown>) => {
   const type = "DNS";
-  if (Array.isArray(fragment.data)) {
-    return fragment.data.map((issuer: Identity) => ({
+  if (DnsTxtVerificationStatusDataV3.guard(data)) {
+    return {
+      identifier: data.identifier,
+      type,
+    };
+  } else if (DnsTxtVerificationStatusArray.guard(data)) {
+    return data.map((issuer) => ({
       identifier: issuer.location,
       type,
     }));
   }
-  return {
-    identifier: fragment.data?.identifier,
-    type,
-  };
+  throw new Error("Fragment for DNS not supported");
 };
 
-const getDnsDidIdentifierProof = (
-  fragment: VerificationFragment<DnsVerificationFragment | DnsVerificationFragment[]>
-) => {
+const getDnsDidIdentifierProof = ({ data }: VerificationFragmentWithData<unknown>) => {
   const type = "DNS-DID";
-  if (Array.isArray(fragment.data)) {
-    return fragment.data.map((issuer: DnsVerificationFragment) => ({
+  if (DnsDidVerificationStatusArray.guard(data)) {
+    return data.map((issuer) => ({
       identifier: issuer.location,
       type,
     }));
+  } else if (DnsDidVerificationStatus.guard(data)) {
+    return {
+      identifier: data.location,
+      type,
+    };
   }
-  return {
-    identifier: fragment.data?.location,
-    type,
-  };
+  throw new Error("Fragment for DNS-DID not supported");
 };
 
-const getDidIdentifierProof = (
-  fragment: VerificationFragment<SignatureVerificationFragment | SignatureVerificationFragment[]>
-) => {
+const getDidIdentifierProof = ({ data }: VerificationFragmentWithData<unknown>) => {
   const type = "DID";
-  if (Array.isArray(fragment.data)) {
-    return fragment.data.map((issuer: SignatureVerificationFragment) => ({
+  if (DidVerificationStatusArray.guard(data)) {
+    return data.map((issuer) => ({
       identifier: issuer.did,
       type,
     }));
+  } else if (DidVerificationStatus.guard(data)) {
+    return {
+      identifier: data.did,
+      type,
+    };
   }
-  return {
-    identifier: fragment.data?.did,
-    type,
-  };
+  throw new Error("Fragment for DID not supported");
 };
 
-const getIdentityProofFragment = (fragments: VerificationFragment[]) => {
+const getIdentityProofFragment = <T extends AllVerificationFragment<unknown>>(fragments: T[]): T | undefined => {
   if (fragments.length < 1) {
     throw new Error("Please provide at least one verification fragment");
   }
   return fragments.find((status) => status.type === "ISSUER_IDENTITY" && status.status === "VALID");
 };
 
-export const getIdentifier = (fragments: VerificationFragment[]) => {
+export const getIdentifier = <T extends AllVerificationFragment<unknown>>(fragments: T[]) => {
```

Comment:
I wonder for legacy reason do we want to combine the identifiers for all the `ISSUER_IDENTITY` fragments? If not its only returning the first which is pretty inaccurate if it's a group of issuers.

Reply:
I can open an issue to address this later

Reply:
done